src/library/dao/playlistdao.cpp

```diff
@@ -227,7 +227,57 @@ void PlaylistDAO::deletePlaylist(const int playlistId) {
     }
 }
 
-int PlaylistDAO::deleteAllUnlockedPlaylistsWithFewerTracks(
+bool PlaylistDAO::deletePlaylists(const QStringList& idStringList) {
+    if (idStringList.isEmpty()) {
+        return false;
+    }
+    const QString idString = idStringList.join(",");
+
+    qInfo() << "Deleting" << idStringList.size() << "playlists";
+
+    // delete tracks assigned to these playlists
+    auto deleteTracks = FwdSqlQuery(m_database,
+            QString("DELETE FROM PlaylistTracks WHERE playlist_id IN (%1)")
+                    .arg(idString));
+    if (!deleteTracks.execPrepared()) {
+        return false;
+    }
+
+    // delete the playlists
+    auto deletePlaylists = FwdSqlQuery(m_database,
+            QString("DELETE FROM Playlists WHERE id IN (%1)").arg(idString));
+    if (!deletePlaylists.execPrepared()) {
+        return false;
+    }
+
+    emit deleted(kInvalidPlaylistId);
+    return true;
+}
+
+bool PlaylistDAO::deleteUnlockedPlaylists(QStringList& idStringList) {
```

Comment:
`idStringList` is not const anymore so we can remove the locked playlist ids, without creating another QStringList

Reply:
We don't want non const references. I think in this case it would be correct to std::move the idStringList into the function, than it is clear that the caller must not use it afterwards.
```suggestion
bool PlaylistDAO::deleteUnlockedPlaylists(QStringList&& idStringList) {
```